cli/azd/pkg/apphost/generate.go

```diff
@@ -420,6 +420,10 @@ func (b *infraGenerator) Compile() error {
 		}
 
 		if binding != nil {
+			if binding.ContainerPort == nil {
+				return fmt.Errorf("binding for %s resource does not have containerPort specified", name)
```

Comment:
Getting my feet wet here, so looking for any guidance around:

1. Should "validation" be in `validateAndMergeBindings` (knowing that projects don't currently use the binding port)? Or is `validateAndMergeBindings` meant to simply be `mergeBindings` with some attached validation?
2. I chose specifically to call out `containerPort` instead of "container port property" because it's mean to be a "oops, a dev error occurred" type situation. What do we think?

Reply:
I think my preference is what you have here (over changing `validateAndMergeBindings`).  The semantics of "validate" in `validateAndMergeBindings` was more of: "Are these bindings similar enough that we can map them to the ACA ingress rules (which only allows http and https) and like doesn't allow you to mix say listening on http and then some random port with just say TCP ingress (because you can't use the ACA L7 load balancer then).

To me this feels like a "the manifest is invalid and would never be correct" (since the containerPort is a required property, per https://github.com/dotnet/aspire/issues/693) vs "this correct manifest can't be deployed to azure, because it describes a binding topology, we can't make real on ACA").  And so I think doing the validation here (or if I was to put my compiler engineer hat on, have a pass in `LoadManifest` that walked over the manifest and did this sort of validation, and perhaps other stuff we find we need).

I do wonder if for now we can make the error a little more actionable for our Aspire friends.  Could we say something like:

"binding form %s resource does not specify a container port, please ensure you pass a `hostPort` value to `WithServiceBinding` for this resource in your Aspire App Host project"?

That gives folks a pointer as to what they need to fix in terms of .NET Primitives they understand as aspire developers, instead of the name of the property in the manifest they may be blissfully unaware of.